frontend/src/core/resource/reducer.js

```diff
@@ -0,0 +1,69 @@
+/* @flow */
+
+import { RECEIVE, UPDATE } from './actions';
+
+import type { Resource, ReceiveAction, UpdateAction } from './actions';
+
+
+type Action =
+    | ReceiveAction
+    | UpdateAction
+;
+
+export type ResourcesState = {|
+    +resources: Array<Resource>,
+|};
+
+
+function updateResource(
+    state: Object,
+    resource_path: string,
+    approved_strings: number,
+    strings_with_warnings: number,
+): Array<Resource> {
+    return state.resources.map(item => {
+        if (item.path === resource_path) {
+            const allResources = state.resources.slice(-1)[0];
+
+            const diff_approved = approved_strings - item.approved_strings;
+            item.approved_strings += diff_approved;
+            allResources.approved_strings += diff_approved;
```

Comment:
`item` and `allResources` should be immutable. You should instead copy these objects first, and recreate a new list of resources based on them. Note that you don't need to copy the ones you do not modify.

Reply:
I am very disappointed that Flow didn't catch that. :-(